client/state/selectors/is-gmb-nudge-visible.js

```diff
@@ -7,18 +7,56 @@ import createSelector from 'lib/create-selector';
 import { getSiteOption, getSitePlanSlug } from 'state/sites/selectors';
 import { PLAN_BUSINESS } from 'lib/plans/constants';
 
+const WEEK_IN_SECONDS = 60 * 60 * 24 * 7;
+
 /**
- * Returns true if the Google My Business (GMB) nudge should be visible in stats
+ * Returns true if site has promote goal set
  *
  * @param  {Object}  state  Global state tree
  * @param  {String}  siteId The Site ID
- * @return {Boolean} True if we should show the nudge
+ * @return {Boolean} True if site has "promote" goal
  */
-export default createSelector(
+const siteHasPromoteGoal = createSelector(
 	( state, siteId ) => {
 		const siteGoals = ( getSiteOption( state, siteId, 'site_goals' ) || '' ).split( ',' );
-		const sitePlan = getSitePlanSlug( state, siteId );
-		return siteGoals.indexOf( 'promote' ) !== -1 && sitePlan === PLAN_BUSINESS;
+		return siteGoals.indexOf( 'promote' ) !== -1;
 	},
-	( state, siteId ) => [ getSitePlanSlug( state ), getSiteOption( state, siteId, 'site_goals' ) ]
+	( state, siteId ) => [ getSiteOption( state, siteId, 'site_goals' ) ]
+);
+
+/**
+ * Returns true if site has business plan
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {String}  siteId The Site ID
+ * @return {Boolean} True if site has business plan
+ */
+const siteHasBusinessPlan = createSelector(
+	( state, siteId ) => getSitePlanSlug( state, siteId ) === PLAN_BUSINESS,
+	( state, siteId ) => [ getSitePlanSlug( state, siteId ) ]
 );
+
+/**
+ * Returns true if the Google My Business (GMB) nudge should be visible in stats
+ *
+ * It should be visible if:
+ * - site is older than 1 week,
+ * - site has a business plan
+ * - site has a promote goal
+ * @param  {Object}  state  Global state tree
+ * @param  {String}  siteId The Site ID
+ * @return {Boolean} True if we should show the nudge
+ */
+const isGmbNudgeVisible = ( state, siteId ) => {
+	const createdAt = getSiteOption( state, siteId, 'created_at' );
+	const isWeekPassedSinceSiteCreation =
+		Date.parse( createdAt ) + WEEK_IN_SECONDS * 1000 < Date.now();
+
+	return (
+		isWeekPassedSinceSiteCreation &&
+		siteHasBusinessPlan( state, siteId ) &&
+		siteHasPromoteGoal( state, siteId )
+	);
+};
```

Comment:
Just a suggestion, but we could avoid unnecessary computations by moving around conditions:

```javascript
const isGmbNudgeVisible = ( state, siteId ) => {
  if ( ! siteHasPromoteGoal( state, siteId ) || ! siteHasBusinessPlan( state, siteId ) ) {
    return false;
  }

  const createdAt = getSiteOption( state, siteId, 'created_at' );

  // Starts showing nudge one week after site creation
  return Date.parse( createdAt ) + WEEK_IN_SECONDS * 1000 < Date.now();
};
```

Reply:
IMO the perf. gain is not that high, while having all of the conditions together is more readable:

```js
isWeekPassedSinceSiteCreation &&
    siteHasBusinessPlan( state, siteId ) &&
    siteHasPromoteGoal( state, siteId )
```